src/SFML/Graphics/RenderTextureImplFBO.cpp

```diff
@@ -153,15 +80,22 @@ RenderTextureImplFBO::~RenderTextureImplFBO()
         glCheck(GLEXT_glDeleteRenderbuffers(1, &depthStencilBuffer));
     }
 
-    // Move all frame buffer objects to stale set
-    for (auto& [contextId, frameBufferId] : m_frameBuffers)
-        staleFrameBuffers.emplace(contextId, frameBufferId);
+    // Unregister FBOs with the contexts if they haven't already been destroyed
+    for (auto& entry : m_frameBuffers)
+    {
+        auto frameBuffer = entry.second.lock();
 
-    for (auto& [contextId, multisampleFrameBufferId] : m_multisampleFrameBuffers)
-        staleFrameBuffers.emplace(contextId, multisampleFrameBufferId);
+        if (frameBuffer)
+            unregisterUnsharedGlObject(frameBuffer);
```

Comment:
```suggestion
            unregisterUnsharedGlObject(std::move(frameBuffer));
```

Reply:
Some recently clang-tidy changes I merged will automatically enforce this once Binary can rebase the PR.